gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt

```diff
@@ -146,18 +146,31 @@ internal class AvatarPickerViewModel(
             when (val result = avatarRepository.uploadAvatar(email, uri)) {
                 is GravatarResult.Success -> {
                     fileUtils.deleteFile(uri)
-                    _uiState.update { currentState ->
-                        val avatar = result.value
-                        currentState.copy(
-                            uploadingAvatar = null,
-                            emailAvatars = currentState.emailAvatars?.copy(
-                                avatars = buildList {
-                                    add(avatar)
-                                    addAll(currentState.emailAvatars.avatars.filter { it.imageId != avatar.imageId })
-                                },
-                            ),
-                            scrollToIndex = null,
-                        )
+                    val isAutoSelected = _uiState.value.emailAvatars?.selectedAvatarId == null
+                    if (isAutoSelected) {
+                        fetchAvatars(showLoading = false, scrollToSelected = true)
+                        _uiState.update { currentState ->
+                            currentState.copy(
+                                uploadingAvatar = null,
+                                avatarUpdates = currentState.avatarUpdates.inc(),
+                            )
+                        }
+                    } else {
```

Comment:
Shouldn't we emit the `AvatarPickerAction.AvatarSelected` here?

Reply:
Good catch! Done!